packages/peregrine/lib/talons/MagentoRoute/helpers.js

```diff
@@ -0,0 +1,20 @@
+const CODE_PERMANENT_REDIRECT = 301;
+const CODE_TEMPORARY_REDIRECT = 302;
+export const REDIRECT_CODES = new Set()
+    .add(CODE_PERMANENT_REDIRECT)
+    .add(CODE_TEMPORARY_REDIRECT);
+
+export const RESPONSES = {
+    ERROR: routeError => ({ hasError: true, routeError }),
+    FOUND: component => component,
+    LOADING: { isLoading: true },
+    NOT_FOUND: { isNotFound: true },
+    REDIRECT: relativeUrl => ({ isRedirect: true, relativeUrl })
+};
+
+export const getRouteKey = (pathname, store) => `[${store}]--${pathname}`;
```

Comment:
Is prepending the store view code just to provide protection in the event the page is not reloaded/refreshed when you switch store views? In practice I found that we always refresh the page so there was never an instance where the local state map would contain values from `default` _and_ `fr`, for example.

My point is essentially that we might be able to do away with the store query in the talon and just use the path as the route key, which would save us a network call.

Reply:
The idea is that when the store view changes, we don't want to assume that the component we have in state is correct, since it could be different.

If you're saying that `MagentoRoute` unmounts and local state is lost when we change stores, then you're right, the logic & call for store code here would be unnecessary. I could remove them, in that case.

Reply:
> If you're saying that MagentoRoute unmounts and local state is lost when we change stores, then you're right, the logic & call for store code here would be unnecessary. I could remove them, in that case.

I am saying just that :) We refresh the page to trigger all the queries again using the new store view code or currency header. There may be a point in the future where we use Apollo's API to trigger requery (`clearStore`?) but that's a future state and we don't need to pre-optimize here, especially when it could get in the way of a render.